src/platform/Linux/bluez/BluezEndpoint.h

```diff
@@ -43,39 +43,72 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-/**
- *    @file
- *          Provides Bluez dbus implementation for BLE
- */
-
 #pragma once
 
+#include <stdint.h>
+
+#include <gio/gio.h>
+#include <glib.h>
+
+#include <ble/CHIPBleServiceData.h>
 #include <lib/core/CHIPError.h>
-#include <system/SystemPacketBuffer.h>
 
+#include "BluezConnection.h"
 #include "Types.h"
 
 namespace chip {
 namespace DeviceLayer {
 namespace Internal {
 
+struct BLEAdvConfig;
+
+struct BluezEndpoint
+{
+    char * mpOwningName; // Bus owning name
+
+    // Adapter properties
+    char * mpAdapterName;
+    char * mpAdapterAddr;
+
+    // Paths for objects published by this service
+    char * mpRootPath;
+    char * mpAdvPath;
+    char * mpServicePath;
+
+    // Objects (interfaces) subscibed to by this service
+    GDBusObjectManager * mpObjMgr = nullptr;
+    BluezAdapter1 * mpAdapter     = nullptr;
+    BluezDevice1 * mpDevice       = nullptr;
+
+    // Objects (interfaces) published by this service
+    GDBusObjectManagerServer * mpRoot;
+    BluezGattService1 * mpService;
+    BluezGattCharacteristic1 * mpC1;
+    BluezGattCharacteristic1 * mpC2;
+    // additional data characteristics
+    BluezGattCharacteristic1 * mpC3;
+
+    // map device path to the connection
+    GHashTable * mpConnMap;
+    uint32_t mAdapterId;
+    bool mIsCentral;
+    char * mpAdvertisingUUID;
+    chip::Ble::ChipBLEDeviceIdentificationInfo mDeviceIdInfo;
+    ChipAdvType mType;  ///< Advertisement type.
+    uint16_t mDuration; ///< Advertisement interval (in ms).
```

Comment:
if this is a time quantity, name should be `mDurationMs` for easier code review and maintenance.